src/main/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporter.java

```diff
@@ -288,11 +300,30 @@ private GreenbidsAdUnit createAdUnit(
                 .orElseGet(() -> GreenbidsUnifiedCode.of(
                         adUnitCode, GreenbidsSource.AD_UNIT_CODE_SOURCE.getValue()));
 
+        final ObjectNode biddersOptional = Optional.ofNullable(impExt)
+                .map(ext -> extImpPrebid(ext.get("prebid")))
+                .map(ExtImpPrebid::getBidder)
+                .orElse(null);
```

Comment:
`Optional.ofNullable(impExt).map(ext -> extImpPrebid(ext.get("prebid")))` is the duplicate, extract prebid once and re-use it